pkgs/build-support/dart/build-dart-application/hooks/dart-fixup-hook.sh

```diff
@@ -0,0 +1,21 @@
+# shellcheck shell=bash
+
+dartFixupHook() {
+    echo "Executing dartFixupHook"
+
+    echo "Providing runtime dependencies"
+    if [[ ! -z "$runtimeDependencyLibraryPath" ]]; then
+        # Add runtime library dependencies to the LD_LIBRARY_PATH.
+        # For some reason, the RUNPATH of the executable is not used to load dynamic libraries in dart:ffi with DynamicLibrary.open().
+        #
+        # This could alternatively be fixed with patchelf --add-needed, but this would cause all the libraries to be opened immediately,
+        # which is not what application authors expect.
```

Comment:
but what we should probably do because LD_LIBRARY_PATH taints all programs start from the wrapped program and potentially causing them to load the wrong libraries

Reply:
This can be done in a later PR - this one is just a refactor.